pkg/component/runtime/state.go

```diff
@@ -212,6 +233,11 @@ func (s *ComponentState) syncUnits(comp *component.Component) bool {
 			}
 		}
 	}
+
+	if !gproto.Equal(s.Features, comp.Features) {
```

Comment:
Why do we need to guard against nil above but not here?

```go
if comp.Features != nil &&
		!gproto.Equal(s.expectedFeatures, comp.Features) {
```

Reply:
Good catch, the `nil` check is not needed.  I checked the tests for `gproto.Equal`: https://github.com/protocolbuffers/protobuf-go/blob/808c66411fe76c839a68168654228446fbdc1ecf/proto/equal_test.go#L20.

Reply:
Removed `nil` check in 1ba0bb8b7.